example/lib/country/country_selection/widget/country_selection_bottom_dialog.dart

```diff
@@ -58,6 +74,10 @@ class _CountrySelectionBottomDialogState
 
     return BlocConsumer<CountrySelectionBloc, CountrySelectionState>(
       listener: (context, state) {
+        if (state.countriesLoaded) {
+          _onCountriesLoaded(countries: state.unsortedCountries);
```

Comment:
If I am not mistaking you call `ClearCountriesLoadedEvent` (later on) to prevent `_onCountriesLoaded()` firing when you update bloc state with sorted countries. Maybe it is worth to separate `_onCountriesLoaded()` and `_onError()` calls into two separate bloc listeners (according to single responsibility principle), so you can use listenWhen on `_onCountriesLoaded()` call something like so:
```
listenWhen: (previousState, state) {
   state.countriesLoaded && previousState.unsortedCountries != state.unsortedCountries
},
listener: (context, state) {
   _onCountriesLoaded(countries: state.unsortedCountries);
},
```

Reply:
You are right, `ClearCountriesLoadedEvent` is called to set `countriesLoaded` to false. I use `countriesLoaded` to fire `_onCountriesLoaded()' on initial load only. Though I don't think that having 2 `BlocListener/Consumer` is a right thing, since it's kinda breaks SRP (because now you observe same state in two different widgets). I believe, that having additional flag like `countriesLoaded` is an okay evil to achieve needed behaviour.